war/src/main/java/io/lumeer/engine/controller/DocumentFacade.java

```diff
@@ -324,10 +284,77 @@ public void dropAttribute(final String collectionName, final String documentId,
     */
    public Set<String> getDocumentAttributes(final String collectionName, final String documentId) throws DbException {
       DataDocument dataDocument = dataStorage.readDocument(collectionName, dataStorageDialect.documentIdFilter(documentId));
-      Set<String> documentAttributes = new HashSet<>();
-      // filter out metadata attributes
-      documentAttributes.addAll(dataDocument.keySet().stream().filter(key -> !key.startsWith(LumeerConst.Document.METADATA_PREFIX)).collect(Collectors.toList()));
-      return documentAttributes;
+      return dataDocument != null ? getDocumentAttributes(dataDocument) : null;
+   }
+
+   private Set<String> getDocumentAttributes(DataDocument dataDocument) {
```

Comment:
Maybe you could add some javadoc or just short comment also to private methods - for future reference.